x/concentrated-liquidity/incentives_test.go

```diff
@@ -3557,9 +3603,19 @@ func (s *KeeperTestSuite) TestGetIncentiveRecordSerialized() {
 // It has been determined that no funds are at risk. The incentives are eventually distributed if either:
 // a) Long time without an update to the pool state occurs (at least 51 minute with the current configuration)
 // b) current tick liquidity becomes smaller
+//
+// As a solution, we used a scaling factor to reduce the likelihood of truncation.
+// This test shows that the scaling factor is effective in reducing the likelihood of truncation.
+// However, the scaling factor does not eliminate the possibility of truncation.
 func (s *KeeperTestSuite) TestIncentiveTruncation() {
 	s.SetupTest()
 
+	// We multiply the current tick liqudity by this factor
+	// To bring the current tick liquidity to be at the border line of truncating.
+	// Then, we choose values on each side of the threshold to show that truncation is still possible
+	// but, with the scaling factor, it is less likely to occur due to more room for liquidity to grow.
+	currentTickLiquidityIncreaseFactor := osmomath.BigDecFromDec(cl.PerUnitLiqScalingFactor)
```

Comment:
Are we able to write out somewhere what exactly these limits are? As in, are we just buying time and if a pool gets 100M liquidity we are back in the same boat, or is the liquidity unlikely (something like 1T for a single pool?)

Reply:
The analysis is in the appendix of the hack md of the PR description.

In a sense, if liquidity growths infinitely, the limit will be hit again.

However, we have alerts in-place. Also, the buffer should be sufficient based on analysis.

Going higher with the scaling factor is also risky because, then, we would be running into overflows